### ISSUE TITLE
Investigation of account bans

### ISSUE THREAD

**Issue by antizhyk** [d60/twikit#14]
Hi everyone, here comes the moment when my Twitter account got blocked. I decided to open this topic and describe what I did and why it happened. I think if other people will describe similar cases here and share their thoughts on it, we can make the application better and beat bans in the future.

I wrote a bot based on this app that wrote messages in group chats and made reposts from other accounts' pages. It activated every 30 minutes, wrote one message in 10 chats, made 30 reposts and went to sleep for 30 minutes.  During the activity, he would pause for about a minute (the time was randomly selected between 30 seconds and 1.5 minutes). After about 10-12, my account logged out of Twitter on its own. I logged back into twitter through my browser and decided to change the cookies in the bot. I copied the cookies from my plugin and pasted them into the bot, launched it and was immediately banned. Looking carefully at the cookies, I found that they had not been updated and it turned out that I had put the same cookies as before (most likely a bug in the cookie reader plugin). I think that's the main reason: if Twitter kicks the account out, the next time I log in from the app, I'll have to log in with new cookies. But we'll see what happens next, I'll write about my further experience here.

**Comment by d60** [comment]
I'm sorry to hear that your bot using my application got banned. To further improve, I'd like to know more about the circumstances under which you were running the bot.

Firstly, what kind of bot were you creating and running? I'm interested in whether it violated Twitter's terms of service or community guidelines.

Secondly, how were you logging into your account? Were you logging in every time using the login method, or were you reusing cookies with methods like save_cookies and load_cookies? If you were logging in every time, that might be one of the reasons for the ban since frequent logins can trigger suspicion from Twitter.

Again, I apologize for any inconvenience this may have caused. Thank you.

**Comment by antizhyk** [comment]
It's no big deal, I knew what I was getting into. I'm trying to create the next program:
1. From the selling account, he writes a random message to the group chat with the same accounts.
2. Then he goes to the page of people who wrote messages in this chat earlier and makes reposts of their media tweets.

These people in return make reposts of our tweets and so there is mutual advertising. 
Here's my code: 

``` py
from client import Client
import logging
import random
import time

client = Client('en-US', proxy_url)
client.load_cookies('./auth_cookies.json')

# Logging setup
logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')

while True:
    try:
        logging.info("Starting chat processing")

        chat_ids = [
            {'id': '1729121949765259721', 'count': 5},
            {'id': '1727362055878197563', 'count': 3},
            # Additional chats can be added here
        ]

        for chat in chat_ids:
            logging.info(f"Processing chat with ID: {chat['id']}")
            messages = client.get_dm_history(chat['id'], None, None, True, True)
            messages = messages[:chat['count']]
            if check_if_posted(client, messages):
                logging.info("Message already sent")
                continue
            # Your message processing logic here
            logging.info("Sending message")
            post_message(my_messages, client, chat['id'])
            sleep_time = random.randint(30, 120)

            for message in messages:
                user = get_account(client, message)
                if not user:
                    logging.warning("User not found")
                    continue

                logging.info(f"Processing user {user.screen_name}")
                is_retweeted_pinned = check_pinned(client, user)
                if is_retweeted_pinned:
                    logging.info("Pinned tweet detected. Skipping.")
                    sleep_time = random.randint(30, 120)
                    continue
                if not is_retweeted_pinned:
                    is_retweeted_media = check_media(user)
                    if is_retweeted_media:
                        logging.info("Media files detected. Skipping.")
                        sleep_time = random.randint(30, 120)
                        continue
    except Exception as e:
        logging.error(f"An error occurred: {e}", exc_info=True)
        if hasattr(e, 'status_code') and e.status in [401, 403]:
            logging.error(f"Access denied or authentication required: {e}")
            break
    finally:
        logging.info("Waiting for 30 minutes...")
        time.sleep(1800)  # Break for 30 minutes
```

**Comment by d60** [comment]
I see. The purpose of this program does not appear to be inherently in violation of any specific terms and conditions, and the rate limiting measures appear to be well taken. Therefore, I cannot speculate as to why your account was banned. If you would like, please create a new account and try again. Also, please let me know if you have any improvements I should make or ideas on how to combat the ban.
Best of luck and I hope to see you again soon!

**Comment by antizhyk** [comment]
Yes of course already created an account, I'll post here how I'm doing and what's going on.

**Comment by d60** [comment]
Hello,
I also tried various things about the DM feature. When I continued sending similar formats to the same group, I received the following warning/error:

`
twikit.errors.Forbidden: status: 403, message: "{"errors":[{"code":226,"message":"This request looks like it might be automated. To protect our users from spam and other malicious activity, we can't complete this action right now. Please try again later."}]}"
`

In other words, Twitter may be enforcing stricter measures against actions related to messages as part of their spam prevention efforts. Therefore, please try changing the message format each time or experimenting with different approaches.
Wishing you luck.

**Comment by d60** [comment]
Hello,

I also tried various things about the DM feature. When I continued sending similar formats to the same group, I received the following warning/error:
`twikit.errors.Forbidden: status: 403, message: "{"errors":[{"code":226,"message":"This request looks like it might be automated. To protect our users from spam and other malicious activity, we can't complete this action right now. Please try again later."}]}"`
In other words, Twitter may be enforcing stricter measures against actions related to messages as part of their spam prevention efforts. Therefore, please try changing the message format each time or experimenting with different approaches.

Best of luck.
